src/main/java/com/ekart/service/ProductService.java

```diff
@@ -44,7 +57,55 @@ public Product getProductById(Long id) throws RecordNotFoundException {
         }
     }
 
-    public Product saveProduct(Product product) {
+    public Product saveProduct(Map<String, Object> request) {
+        Category newCategory;
+        Brand newBrand;
+        Set<Size> newSize = new HashSet<>();
+        List<String> desc = new ArrayList<>();
+
+        String categoryName = (String) request.get(Constants.CATEGORY);
+        Optional<Category> category = categoryRepository.findCategoryByName(categoryName);
+        if(!category.isPresent()) {
```

Comment:
I think if category, brand, or size is not present then we should throw an error. We should not add it.

We should not allow adding these while saving products. Separate endpoints and (in ui) dialog should be present to do that.

Reply:
Correct. For now let it be it will allow us to insert all categories, brand etc to db. Once we have enough data for testing we can get rid of this

Reply:
ok